### PR TITLE
HHH-11131 - Fix exporting Envers generated ordered sequence for Oracle databases.

### PR DESCRIPTION
**Pull request by Naros**
https://hibernate.atlassian.net/browse/HHH-11131

### CODE REVIEW

File: hibernate-envers/src/main/java/org/hibernate/envers/enhanced/OrderedSequenceGenerator.java [hunk 1]
```diff
@@ -6,25 +6,37 @@
...
+ * @author Chris Cranford
  */
 public class OrderedSequenceGenerator extends SequenceStyleGenerator {
 	@Override
-	public String[] sqlCreateStrings(Dialect dialect) throws HibernateException {
-		String[] create = super.sqlCreateStrings( dialect );
-		if ( dialect instanceof Oracle8iDialect ) {
-			// Make sure that sequence produces increasing values in Oracle RAC environment.
-			create = StringHelper.suffix( create, " order" );
+	protected DatabaseStructure buildDatabaseStructure(
```
Comment (sebersole) [inline hibernate-envers/src/main/java/org/hibernate/envers/enhanced/OrderedSequenceGenerator.java]: I'd maybe split this out in the base class into `#buildSequenceStructure` and `#buildTableStructure`.  That allows to not duplicate the `jdbcEnvironment.getDialect().supportsSequences() && !forceTableUse` check

File: hibernate-envers/src/main/java/org/hibernate/envers/enhanced/OrderedSequenceStructure.java [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+import org.hibernate.id.enhanced.SequenceStructure;
+import org.hibernate.internal.CoreMessageLogger;
+import org.jboss.logging.Logger;
+
+/**
+ * @author Chris Cranford
+ */
+public class OrderedSequenceStructure implements DatabaseStructure {
+
+	// @todo: a fair bit of duplication from SequenceStructure - needs refactor
```
Comment (sebersole) [inline hibernate-envers/src/main/java/org/hibernate/envers/enhanced/OrderedSequenceStructure.java]: +1 - this should be refactored

File: hibernate-envers/src/main/java/org/hibernate/envers/enhanced/OrderedSequenceGenerator.java (hunk 1 above)
Comment (Naros) [inline hibernate-envers/src/main/java/org/hibernate/envers/enhanced/OrderedSequenceGenerator.java]: @sebersole done.

File: hibernate-envers/src/main/java/org/hibernate/envers/enhanced/OrderedSequenceStructure.java (hunk 2 above)
Comment (Naros) [inline hibernate-envers/src/main/java/org/hibernate/envers/enhanced/OrderedSequenceStructure.java]: @sebersole done.

**Comment by dreab8** [comment]
merged upstream

**Pull request closed without merging**